package.json

```diff
@@ -18,19 +23,14 @@
     "shallow-equals": "1.0.0"
   },
   "devDependencies": {
-    "babel-cli": "6.26.0",
```

Comment:
I'm in favour of it, it's required to build so if we don't put it under regular dependencies if someone wants to use mattermost-redux they need to install AND install babel. npm and yarn are specifically built so that you don't need to do things like that

Reply:
one could argue that the same applies to the webapp, etc.. I'm 2/5 in keeping it where it is, normally the dev dependencies won't get installed if you set your NODE_ENV to production, and I haven't had any issues using it the way it is so far

Reply:
Because mattermost-redux is a npm module for other projects.

For instance, when I use this module as an API client library in my project.

That will just fail the postinstall until I finish the setup for Babel.

If babel moved, postinstall succeeds, even though I don't have a babel in my project, and I can use it right in the project. This is because dependencies are installed together when the npm module is installed.

Reply:
^ Exactly. And the webapp will never be a module used by another project so it doesn't apply there

Reply:
yes but then you would want to use the published module which should be already transpiled

Reply:
[prepublish was deprecated by npm](https://docs.npmjs.com/misc/scripts#prepublish-and-prepare) as I know.

Reply:
@jwilander correct me if I'm wrong but the plan is publish this library in npm so people can use it with `npm install mattermost-redux` right?

If I'm not mistaken when you publish the package to npm it will be already in its transpiled form or will it be exactly as it is in the repo?

if it is in the transpiled form then there's not need to do this at all, if it is the second option then it would make sense to have it as in this PR.

Or I'm totally missing the point here

Reply:
https://github.com/CometKim/mattermost-redux/blob/node/.babelrc

And this is what actually I use in my project for less polyfills. (for Node 8+)

Reply:
I'm not entirely sure, but if we build before publishing I think that's the case

Reply:
Exactly so that’s my whole point if we build the module and then publish it no-one will need to use babel to transpile it so it doesn’t make sense to have babel as a dependency instead of a dev dependency

Reply:
Babel is only for "transpile" anyway. you should not need it when you use it as npm package.
like when you use it, what you do is just `const module = require("module")` which doesn't care about babel or anything at all.
also even if you publish this "with" babel dependency, that doesn't means anything from the side that using the module. (it just install babel as a dependency, which is unnecessary at all.) again, it's Transpiler. you must use compiled JS not original code when you publish it. same applies to TS or Coffee or ES6.. anything. checkout any TS based libraries if you don't get it

Reply:
@breath103 @enahum Yes. You right, Babel is only for transpile. The build script is now postpublish and all other projects which use this module is actually depend on babel for now.

So, should be changed to `prepublishOnly`. (and I can't use this before you publish to npm. or build manually)

I'm totally OK for this for webapp.
However, for node, I think using postpublish + env preset can provide more flexible polyfills for actual user's node environment.